modules/scripts-utils/ace-adapter/src/main/java/org/daisy/pipeline/epub/AceProvider.java

```diff
@@ -0,0 +1,181 @@
+package org.daisy.pipeline.epub;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import net.sf.saxon.s9api.QName;
+import net.sf.saxon.s9api.SaxonApiException;
+
+import org.daisy.common.xproc.calabash.XProcStepProvider;
+
+import com.xmlcalabash.core.XProcException;
+import com.xmlcalabash.core.XProcRuntime;
+import com.xmlcalabash.core.XProcStep;
+import com.xmlcalabash.io.WritablePipe;
+import com.xmlcalabash.library.DefaultStep;
+
+import com.xmlcalabash.runtime.XAtomicStep;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.daisy.common.shell.CommandRunner;
+import org.daisy.common.shell.BinaryFinder;
+
+import java.util.Optional;
+
+
+/**
+ * Accessibilty Checker for Epub tools validator provider : 
+ * This class provide a step that tries to validate an epub using
+ * a system-wide install of the ace tools if it is installed.
+ */
+@Component(
+	name = "pxi:ace",
+	service = { XProcStepProvider.class },
+	property = { "type:String={http://www.daisy.org/ns/pipeline/xproc/internal}ace" }
+)
+public class AceProvider implements XProcStepProvider {
+	
+	
+	
+	public XProcStep newStep(XProcRuntime runtime, XAtomicStep step) {
+		return new AceStep(runtime, step);
+	}
+	
+	private static File aceProgram = null;
+	
+	/**
+	 * Initialize the provider :
+	 * Retrieve the Ace executable path
+	 * @throws RuntimeException if Ace is not found on the system
+	 */
+	@Activate
+	public void init() {
+		Optional<String> lpath = BinaryFinder.find("ace");
+		if(lpath.isPresent()) {
+			aceProgram = new File(lpath.get());
+		} else throw new RuntimeException("ACE was not found on your system");
```

Comment:
I'm not sure whether the exception should be thrown here or below in the `run` method. But probably not both.